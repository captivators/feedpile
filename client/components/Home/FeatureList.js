import React from 'react';
import Feature from './Feature'

const style = {
  first: {
    paddingTop: '45px',
    paddingBottom: '45px',
    width: '100%',
    textAlign: 'center',
    h1: {
      color: '#fff',
      marginTop: 0,
      fontSize: '48px',
    },
  },
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  }
}

const featureList = [
  {
    name: 'RSS',
    featureName: 'Seamless updates',
    description: 'Receive the newest articles instantaneously'
  },
  {
    name: 'News',
    featureName: 'Access to any feed',
    description: 'Unlimited resources all at your fingertips'
  }
]

const FeatureList = (props) => {
  console.log(props)
  return (
    <div style={style.first}>
      <h2>Features</h2>
      <div className="team" style={style.root}>
        {featureList.map((feature, index) => (
          <Feature info={ feature } key={ index } />
        ))}
      </div>
    </div>
    )
}

export default FeatureList;