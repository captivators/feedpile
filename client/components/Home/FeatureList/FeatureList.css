.features {
  display: flex;
  margin: auto;
  width: 100%;
  text-align: center;
  background-color: #B0BEC5;
  justify-content: space-between;
  padding-top: 80px;
  color: #000;
}

.description {
  flex-grow: 1;
  text-align: justify;
  text-justify: inter-word;
  max-width: auto;
  line-height: 130%;
  padding: 20px;
}

.features-list {
  margin: auto;
  max-width: 90%;
  justify-content: space-between;
  padding-bottom: 80px;
}


.feature-item {
    display: flex;
    flex-wrap: wrap;
    width: auto;
    align-items: flex-start;
    width: 100%;
    justify-content: space-around;
}

.info-a {
  order: 1;
  max-width: 600px;
  padding: 10px;
  width: auto;
}


 .info-b {
  order: 0;
  max-width: 600px;
  min-width: 200px;
  padding: 10px;
  align-self: flex-reverse;
  width: auto;
}

.icon-a {
  order: 0;
  align-self: center;
  margin: auto;
}

.icon-b {
  order: 2;
  align-self: center;
}

@media screen and (max-width: 480px) {
  .icon-b {
    order: 0;
  }
  .info-b {
    order: 1;
  }
}

@media screen and (max-width: 600px) {
  .icon-b {
    text-align: center;
    order: 0;
    margin: auto;
  }
  .info-b {
    order: 1;
  }
}
