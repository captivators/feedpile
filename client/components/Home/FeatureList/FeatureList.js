import React from 'react';
import './FeatureList.css'
import RssFeed from 'material-ui-icons/RssFeed';
import PersonPin from 'material-ui-icons/PersonPin'
import StayCurrentPortrait from 'material-ui-icons/StayCurrentPortrait'
import Language from 'material-ui-icons/Language'

const featureList = [
  {
    name: 'Syndication',
    description: 'Keeping on top of the news across various sites is a hassle. Feedpile uses aggregation technology to consolidate many news outlets into a one-page application, saving you time and energy.',
    icon: <RssFeed style={{
      width: "60px",
      height: "60px",
      marginTop: '50px'
    }}/>,
  },
  {
    name: 'Minimalist Design',
    description: 'FeedPile offers a clean and uncluttered UI to make reading the news a breath of fresh air. Enjoy reading individual articles without a flurry of advertisements.',
    icon: <StayCurrentPortrait style={{
      width: "60px",
      height: "60px",
      order: "2",
      marginTop: '50px'
    }}/>,
  },
  {
    name: 'Customization',
    description: 'Escape the monotonous echo chamber of mainstream media and curate your own personal newspaper easily by subscribing to content from your favorite sites. Store collections of feeds in categories to customize your content.',
    icon: <PersonPin style={{
      width: "60px",
      height: "60px",
      order: "0",
      marginTop: '50px'
    }}/>,
  },
  {

  }
]

const FeatureList = (props) => {
  return (
    <div className="features">
      <ul className="features-list">
        <li>
          <div className="feature-item">
            <div className="icon-a">
              <i>{featureList[0].icon}</i>
            </div>
            <div className="info-a">
              <ul >
                <li style={{fontSize: "24px"}}>{featureList[0].name}</li>
                <br/>
                <li className="description">{featureList[0].description}</li>
              </ul>
            </div>
          </div>
        </li>
        <li>
          <div className="feature-item">
            <div className="icon-b">
            <i>{featureList[1].icon}</i>
            </div>
          <div className="info-b">
          <ul >
          <li style={{fontSize: "24px"}}>{featureList[1].name}</li>
          <br/>
          <li className="description">{featureList[1].description}</li>
          </ul>
          </div>

        </div>
        </li>

        <li>
          <div className="feature-item">
          <div className="icon-a">
          <i>{featureList[2].icon}</i>
          </div>
          <div className="info-a">
          <ul >
          <li style={{fontSize: "24px"}}>{featureList[2].name}</li>
          <br/>
          <li className="description">{featureList[2].description}</li>
          </ul>
          </div>

        </div>
        </li>

      </ul>
    </div>
    )
}

export default FeatureList;
