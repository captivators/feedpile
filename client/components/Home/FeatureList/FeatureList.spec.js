import React from 'react';
import FeatureList from './FeatureList'
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Landingpage: Feature list snapshot test', () => {

  const component = shallow(
    <FeatureList />
    );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});