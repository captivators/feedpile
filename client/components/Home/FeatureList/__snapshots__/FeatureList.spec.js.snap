// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landingpage: Feature list snapshot test 1`] = `
<div
  className="features"
>
  <ul
    className="features-list"
  >
    <li>
      <div
        className="feature-item"
      >
        <div
          className="icon-a"
        >
          <i>
            <pure(RssFeed)
              style={
                Object {
                  "height": "60px",
                  "marginTop": "50px",
                  "width": "60px",
                }
              }
            />
          </i>
        </div>
        <div
          className="info-a"
        >
          <ul>
            <li
              style={
                Object {
                  "fontSize": "24px",
                }
              }
            >
              Syndication
            </li>
            <br />
            <li
              className="description"
            >
              Keeping abreast of the news is demanding and visiting multiple sites frequently to find out if content has been updated is a waste of time. Feedpile uses aggregation technology to consolidate many news outlets into a one page application, saving you time and energy.
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <div
        className="feature-item"
      >
        <div
          className="icon-b"
        >
          <i>
            <pure(StayCurrentPortrait)
              style={
                Object {
                  "height": "60px",
                  "marginTop": "50px",
                  "order": "2",
                  "width": "60px",
                }
              }
            />
          </i>
        </div>
        <div
          className="info-b"
        >
          <ul>
            <li
              style={
                Object {
                  "fontSize": "24px",
                }
              }
            >
              Responsive Design
            </li>
            <br />
            <li
              className="description"
            >
              Our responsive design provides you not only with the sleakest desktop platform but also a clean RSS feed reader for any mobile device.
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <div
        className="feature-item"
      >
        <div
          className="icon-a"
        >
          <i>
            <pure(PersonPin)
              style={
                Object {
                  "height": "60px",
                  "marginTop": "50px",
                  "order": "0",
                  "width": "60px",
                }
              }
            />
          </i>
        </div>
        <div
          className="info-a"
        >
          <ul>
            <li
              style={
                Object {
                  "fontSize": "24px",
                }
              }
            >
              Customization
            </li>
            <br />
            <li
              className="description"
            >
              Escape the monotonous echo chamber of mainstream media and curate your own personal newspaper easily by subscribing to content from you favorite sites. Store collections of feeds in categories to customize your content. Save articles with Evernote integration and share them over social media.
            </li>
          </ul>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
